index.bs

```diff
@@ -3405,6 +3420,56 @@ and |session|:
 
 </div>
 
+<div algorithm="locate nodes using accessibility attributes">
+To <dfn>locate nodes using accessibility attributes</dfn> with given |context nodes|, |selector|,
+|maximum returned node count|, |session|, and |returned nodes|:
+
+1. If |selector|'s <code>role</code> and |selector|'s <code>name</code> are missing, return [=error=] with [=error code=]
```

Comment:
It technically doesn't matter, but I don't love this organisation where we recursively call into the algorithm and re-perform steps that are actually redundant.

I think I'd prefer that we pull out the parts that actually make sense to run in each recursive call into their own algorithm and just call that.